Can't export all members of mutable.TreeMap (tried with List as well)

dottyVersion = "0.27.0-RC1"
Please find 3 different verions and 2 different outputs here:
## Minimized code
1:
```Scala
class MaxSizeMap[K, V](maxSize: Int)(using o: Ordering[K]):
  private val sortedMap: mutable.TreeMap[K, V] = mutable.TreeMap.empty[K, V](o)
  
  export sortedMap._
```
2:
```Scala
class MaxSizeMap[K, V](maxSize: Int)(using o: Ordering[K]):
  val sortedMap: mutable.TreeMap[K, V] = mutable.TreeMap.empty[K, V](o)
  
  export sortedMap._
```
3:
notice that all the methods I had to exclude to get rid of error messages from 1 return this.type
```Scala
class MaxSizeMap[K, V](maxSize: Int)(using o: Ordering[K]):
  private val sortedMap: mutable.TreeMap[K, V] = mutable.TreeMap.empty[K, V](o)
  
  export sortedMap.{mapValuesInPlace => _, toIterable => _, seq => _, --= => _, ++= => _, += => _, addAll => _, subtractAll => _, 
    lazyZip => _, filterInPlace => _, -= => _, addOne => _, subtractOne => _, transform => _, retain => _, _}
```

## Output
1:
```scala
[error] 8 |  export sortedMap._
[error]   |                   ^
[error]   |non-private method += in class MaxSizeMap refers to private value sortedMap
[error]   |in its type signature (elem1: (K, V), elem2: (K, V), elems: (K, V)*): 
[error]   |  (MaxSizeMap.this.sortedMap : collection.mutable.TreeMap[K, V])
```
2 and 3 (same stack trace):
truncated stack trace:
```scala
[error] ## Exception when compiling 6 sources to /home/michal/programming/scala/crypto-db/target/scala-0.27/classes
[error] java.lang.AssertionError: assertion failed
[error] dotty.DottyPredef$.assertFail(DottyPredef.scala:16)
[error] dotty.tools.dotc.transform.ResolveSuper.transformDefDef(ResolveSuper.scala:63)
[error] dotty.tools.dotc.transform.MegaPhase.goDefDef(MegaPhase.scala:973)
[error] dotty.tools.dotc.transform.MegaPhase.goDefDef(MegaPhase.scala:974)
[error] dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:251)
[error] dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:426)
[error] dotty.tools.dotc.transform.MegaPhase.transformStat$2(MegaPhase.scala:436)
[error] dotty.tools.dotc.transform.MegaPhase.recur$3$$anonfun$1(MegaPhase.scala:441)
[error] dotty.tools.dotc.transform.MegaPhase$$Lambda$6255/00000000480AB540.apply(Unknown Source)
[error] scala.collection.immutable.List.mapConserve(List.scala:472)
```
## Expectation
It compiles

--- a/compiler/src/dotty/tools/dotc/core/Flags.scala
+++ b/compiler/src/dotty/tools/dotc/core/Flags.scala
@@ -517,6 +517,8 @@ object Flags {
   val AccessorOrDeferred: FlagSet            = Accessor | Deferred
   val PrivateAccessor: FlagSet               = Accessor | Private
   val AccessorOrSynthetic: FlagSet           = Accessor | Synthetic
+  val JavaOrPrivateOrSynthetic: FlagSet      = Artifact | JavaDefined | Private | Synthetic
+  val PrivateOrSynthetic: FlagSet            = Artifact | Private | Synthetic
   val EnumCase: FlagSet                      = Case | Enum
   val CovariantLocal: FlagSet                = Covariant | Local                              // A covariant type parameter
   val ContravariantLocal: FlagSet            = Contravariant | Local                          // A contravariant type parameter
@@ -540,7 +542,6 @@ object Flags {
   val InlineByNameProxy: FlagSet             = InlineProxy | Method
   val JavaEnumTrait: FlagSet                 = JavaDefined | Enum                             // A Java enum trait
   val JavaEnumValue: FlagSet                 = JavaDefined | EnumValue                        // A Java enum value
-  val JavaOrPrivateOrSynthetic: FlagSet      = JavaDefined | Private | Synthetic
   val StaticProtected: FlagSet               = JavaDefined | JavaStatic | Protected           // Java symbol which is `protected` and `static`
   val JavaModule: FlagSet                    = JavaDefined | Module                           // A Java companion object
   val JavaInterface: FlagSet                 = JavaDefined | NoInits | Trait
@@ -561,7 +562,6 @@ object Flags {
   val ValidForeverFlags: FlagSet             = Package | Permanent | Scala2SpecialFlags
   val TermParamOrAccessor: FlagSet           = Param | ParamAccessor
   val PrivateParamAccessor: FlagSet          = ParamAccessor | Private
-  val PrivateOrSynthetic: FlagSet            = Private | Synthetic
   val PrivateOrArtifact: FlagSet             = Private | Artifact
   val ClassTypeParam: FlagSet                = Private | TypeParam
   val Scala2Trait: FlagSet                   = Scala2x | Trait

--- a/compiler/src/dotty/tools/dotc/transform/ResolveSuper.scala
+++ b/compiler/src/dotty/tools/dotc/transform/ResolveSuper.scala
@@ -60,7 +60,7 @@ class ResolveSuper extends MiniPhase with IdentityDenotTransformer { thisPhase =
   override def transformDefDef(ddef: DefDef)(using Context): Tree = {
     val meth = ddef.symbol.asTerm
     if (meth.isSuperAccessor && !meth.is(Deferred)) {
-      assert(ddef.rhs.isEmpty)
+      assert(ddef.rhs.isEmpty, ddef.symbol)
       val cls = meth.owner.asClass
       val ops = new MixinOps(cls, thisPhase)
       import ops._

--- a/compiler/src/dotty/tools/dotc/typer/Namer.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Namer.scala
@@ -1029,10 +1029,13 @@ class Namer { typer: Typer =>
 
         def addWildcardForwarders(seen: List[TermName], span: Span): Unit =
           for mbr <- path.tpe.membersBasedOnFlags(required = EmptyFlags, excluded = PrivateOrSynthetic) do
-            val alias = mbr.name.toTermName
-            if !seen.contains(alias)
-               && mbr.matchesImportBound(if mbr.symbol.is(Given) then givenBound else wildcardBound)
-            then addForwarder(alias, mbr, span)
+            if !mbr.symbol.isSuperAccessor then
+              // Scala 2 superaccessors have neither Synthetic nor Artfact set, so we
+              // need to filter them out here (by contrast, Scala 3 superaccessors are Artifacts)
+              val alias = mbr.name.toTermName
+              if !seen.contains(alias)
+                && mbr.matchesImportBound(if mbr.symbol.is(Given) then givenBound else wildcardBound)
+              then addForwarder(alias, mbr, span)
 
         def addForwarders(sels: List[untpd.ImportSelector], seen: List[TermName]): Unit = sels match
           case sel :: sels1 =>

--- a/tests/pos/i9967.scala
+++ b/tests/pos/i9967.scala
@@ -1,0 +1,6 @@
+import collection.mutable
+
+class MaxSizeMap[K, V](maxSize: Int)(using o: Ordering[K]):
+  val sortedMap: mutable.TreeMap[K, V] = mutable.TreeMap.empty[K, V](o)
+
+  export sortedMap._

